src/Components/Common/prescription-builder/InvestigationBuilder.tsx

```diff
@@ -29,12 +29,28 @@ export interface InvestigationBuilderProps<T> {
   setInvestigations: React.Dispatch<React.SetStateAction<T[]>>;
 }
 
-export default function InvestigationBuilder(
-  props: InvestigationBuilderProps<InvestigationType>,
-) {
-  const { investigations, setInvestigations } = props;
-  const [investigationsList, setInvestigationsList] = useState<string[]>([]);
-  const [activeIdx, setActiveIdx] = useState<number | null>(null);
+export const loadInvestigations = async () => {
+  const fetchInvestigations = async () => {
+    const { data } = await request(routes.listInvestigations);
+    return (
+      data?.results.map(
+        (investigation) =>
+          `${investigation.name} -- ${humanizeStrings(
+            investigation.groups.map((group) => ` ( ${group.name} ) `),
+          )}`,
+      ) ?? []
+    );
+  };
+
+  const fetchInvestigationGroups = async () => {
+    const { data } = await request(routes.listInvestigationGroups);
+    return data?.results.map((group) => `${group.name} (GROUP)`) ?? [];
+  };
+
+  const invs = await fetchInvestigations();
+  const groups = await fetchInvestigationGroups();
```

Comment:
can we use useQuery here, it would simplify it a lot or lets warp line 50 and 51 in a useMemo

Reply:
@khavinshankar but this is a normal function, can't use hooks here right? 😅

Reply:
Oh my bad!